src/common/transformations/src/transformations/faster_hash.cpp

```diff
@@ -0,0 +1,422 @@
+        } else if (const auto& a_ps = ov::as_type<ov::AttributeAdapter<ov::PartialShape>>(&adapter)) {
+            const auto& attrs = a_ps->get();
+            std::stringstream shape_str_stream;
+            shape_str_stream << attrs;
+            m_hash = hash_combine(m_hash, shape_str_stream.str());
+        } else if (const auto& a_dim = ov::as_type<ov::AttributeAdapter<ov::Dimension>>(&adapter)) {
+            const auto& attrs = a_dim->get();
+            std::stringstream dim_str_stream;
+            dim_str_stream << attrs;
+            m_hash = hash_combine(m_hash, dim_str_stream.str());
+            auto dim_str = dim_str_stream.str();
+        } else {
+            // Unsupported attribute type will not be taken into account for hash calculation
+        }
+    }
+
+    void on_adapter(const std::string& name, ov::ValueAccessor<bool>& adapter) override {
+        m_hash = hash_combine(m_hash, name);
+        m_hash = hash_combine(m_hash, adapter.get());
+    }
+    void on_adapter(const std::string& name, ov::ValueAccessor<std::string>& adapter) override {
+        m_hash = hash_combine(m_hash, name);
+        m_hash = hash_combine(m_hash, adapter.get());
+    }
+    void on_adapter(const std::string& name, ov::ValueAccessor<int64_t>& adapter) override {
+        m_hash = hash_combine(m_hash, name);
+        m_hash = hash_combine(m_hash, adapter.get());
+    }
+    void on_adapter(const std::string& name, ov::ValueAccessor<double>& adapter) override {
+        m_hash = hash_combine(m_hash, name);
+        m_hash = hash_combine(m_hash, adapter.get());
+    }
+    void on_adapter(const std::string& name, ov::ValueAccessor<std::vector<int>>& adapter) override {
+        m_hash = hash_combine(m_hash, name);
+        join_hash(m_hash, adapter.get());
+    }
+    void on_adapter(const std::string& name, ov::ValueAccessor<std::vector<int64_t>>& adapter) override {
+        m_hash = hash_combine(m_hash, name);
+        join_hash(m_hash, adapter.get());
+    }
+    void on_adapter(const std::string& name, ov::ValueAccessor<std::vector<uint64_t>>& adapter) override {
+        m_hash = hash_combine(m_hash, name);
+        join_hash(m_hash, adapter.get());
+    }
+    void on_adapter(const std::string& name, ov::ValueAccessor<std::vector<float>>& adapter) override {
+        m_hash = hash_combine(m_hash, name);
+        join_hash(m_hash, adapter.get());
+    }
+    void on_adapter(const std::string& name, ov::ValueAccessor<std::vector<std::string>>& adapter) override {
+        m_hash = hash_combine(m_hash, name);
+        join_hash(m_hash, adapter.get());
+    }
+    void on_adapter(const std::string& name, ov::ValueAccessor<std::shared_ptr<Function>>& adapter) override {
+        m_hash = hash_combine(m_hash, name);
+        model_2_hash(m_hash, *adapter.get());
+    }
+};
+
+template <typename T>
+bool is_name_auto_generated(const T& n) {
+    return n.get_friendly_name() == n.get_name();
+}
```

Comment:
More precisely to check that `m_friendly_name` field is empty.
@ilyachur I think we need to introduce smth like `node.friendly_name_is_set()`

Reply:
I don't want to introduce new public API which is not needed for the end user.

Reply:
So let's make it private :)